src/app/shared/components/template/components/drawer/drawer.component.ts

```diff
@@ -7,22 +7,20 @@ import { TemplateBaseComponent } from "../base";
   templateUrl: "./drawer.component.html",
   styleUrls: ["./drawer.component.scss"],
 })
-export class TmplDrawerComponent extends TemplateBaseComponent implements OnInit, AfterViewInit {
-  @ViewChild("drawer", { read: ElementRef }) drawer: ElementRef;
+export class TmplDrawerComponent extends TemplateBaseComponent implements OnInit {
+  @ViewChild("drawer", { read: ElementRef, static: false }) drawer: ElementRef;
```

Comment:
question
I've never used the 'read' property on viewchild, what does it do in this example?
Also given that the draw is only read during toggle do we need to mark as static: false? (again, out of interest, haven't tested myself)

Reply:
It seems the "read" property is used to specify what to return in the case that the selector has some ambiguity. The ambiguity could be the element is an Angular component, in which case `@ViewChild()` could return the component instance or the ElementRef instance (explained [here](https://stackoverflow.com/a/37476195)), or "read" can be used to select a specific provider token (explained [here](https://blog.logrocket.com/understanding-the-viewchild-and-viewchildren-decorators-in-angular-10/#:~:text=modify%20embedded%20views.-,Using%20the%20read%20property,-The%20read%20property), I'm not sure I fully understand without having a reason to use it like this). 

Either way, I think it's superfluous here so I've removed it.

And I think you're right about `static: false`. I added it when I was still trying to get the element's height in `ngAfterViewInit()`, hoping that it would mean that the value would update after all the `*ngFor` elements had rendered, but this was not the case. In fact it's the default value anyway since Angular 9 apparently.

Reply:
Nice! yeah that sorta makes sense, and also makes sense that it isn't required for this use-case anymore